packages/local_auth/local_auth_android/android/src/main/java/io/flutter/plugins/localauth/LocalAuthPlugin.java

```diff
@@ -147,79 +147,104 @@ private void authenticate(MethodCall call, final Result result) {
     }
 
     authInProgress.set(true);
-    AuthCompletionHandler completionHandler =
-        new AuthCompletionHandler() {
-          @Override
-          public void onSuccess() {
-            authenticateSuccess(result);
-          }
-
-          @Override
-          public void onFailure() {
-            authenticateFail(result);
-          }
+    AuthCompletionHandler completionHandler = createAuthCompletionHandler(result);
 
-          @Override
-          public void onError(String code, String error) {
-            if (authInProgress.compareAndSet(true, false)) {
-              result.error(code, error, null);
-            }
-          }
-        };
-
-    // if is biometricOnly try biometric prompt - might not work
     boolean isBiometricOnly = call.argument("biometricOnly");
-    if (isBiometricOnly) {
-      if (!canAuthenticateWithBiometrics()) {
-        if (!hasBiometricHardware()) {
-          completionHandler.onError("NoHardware", "No biometric hardware found");
-        }
-        completionHandler.onError("NotEnrolled", "No biometrics enrolled on this device.");
-        return;
-      }
-      authHelper =
-          new AuthenticationHelper(
-              lifecycle, (FragmentActivity) activity, call, completionHandler, false);
-      authHelper.authenticate();
-      return;
-    }
+    ArrayList<String> authenticationMethods = getEnrolledBiometrics();
 
-    // API 29 and above
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-      authHelper =
-          new AuthenticationHelper(
-              lifecycle, (FragmentActivity) activity, call, completionHandler, true);
-      authHelper.authenticate();
+    // Handle case where any biometrics are available
+    if (canAuthenticateWithBiometrics()) {
+      if (!isBiometricOnly && canAuthenticateWithDeviceCredential()) {
+        authenticationMethods.add("deviceCredential");
+      }
+      sendAuthenticationRequest(call, completionHandler, authenticationMethods);
       return;
     }
 
-    // API 23 - 28 with fingerprint
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && fingerprintManager != null) {
```

Comment:
This is covered by `BiometricPrompt` now. See [documentation](https://developer.android.com/reference/android/hardware/fingerprint/FingerprintManager).